### ISSUE TITLE
Checkout: Country code autodetection in phone number field is broken

### ISSUE THREAD

**Issue by sirbrillig** [Automattic/wp-calypso#85800]
The phone number field in checkout (used primarily by the domain contact fields) has its own country code field that is different from the normal country code field in checkout. That field is used to determine which formatting pattern to use for the number and also to hide the dialing code (eg: `+1`).

<img src="https://uploads.linear.app/f629fcde-6f9b-48e8-8af3-60eed8823bff/931746d5-eefe-47b1-88ec-939a02969039/d0425750-4f69-4ac6-810b-5215ad72f950?signature=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJwYXRoIjoiL2Y2MjlmY2RlLTZmOWItNDhlOC04YWYzLTYwZWVkODgyM2JmZi85MzE3NDZkNS1lZWZlLTQ3YjEtODhlYy05MzlhMDI5NjkwMzkvZDA0MjU3NTAtNGY2OS00YWM2LTgxMGItNTIxNWFkNzJmOTUwIiwiaWF0IjoxNzg3NzczMjk0LCJleHAiOjE4MTkzNDM4NTR9.ujee5w6cvx1087xXj7uHt9qYgYVbqazrfizUnyyAozg " alt="Screenshot 2026-08-26 at 3.26.45 PM.png" width="491" data-linear-height="165" />

If the country is not manually selected, it should auto-detect the country like this:

[https://user-images.githubusercontent.com/18705930/219798541-d6b1a359-d1f7-45cb-86ba-b4fcc7f1dc9b.mov](<https://user-images.githubusercontent.com/18705930/219798541-d6b1a359-d1f7-45cb-86ba-b4fcc7f1dc9b.mov>)

But it doesn't seem to work, possibly broken by [https://github.com/Automattic/wp-calypso/pull/77920](<https://github.com/Automattic/wp-calypso/pull/77920>)

The result is that the user needs to select the country code manually.

To reproduce:

1. Visit checkout with a domain product in the cart.
2. In the contact info step, enter the following into the Phone field: `+355`

Expected: the +3 should (maybe?) disappear and the country drop-down should switch to Albania.

Actual: the country stays at US.

**Comment by rickmgithub** [comment]
## STEPS ##
1. Go to Upgrades->Domains->Add a Domain
2. Follow the steps through until the checkout
3. On the checkout page is the domain contact information
4. Either click "edit" or see the open form
5. In the "phone" section, the OP reports the country flag is not selecting correctly based on where the person is and a number such as +1 234 4556 needs to be entered for the flag to change


## 📌 REPRODUCTION RESULTS ##
- Tested on Simple – Could Not Replicate 
- Tested on Atomic – Could Not Replicate


## 📌 ACTIONS ##
- Author feedback
- Closed as could not replicate (please reopen if the issue reoccurs) 

##Message to author##
Thanks so much for the report. I couldn't replicate the issue. Are you still able to produce this? Feel free to reopen if you can still replicate.

**Comment by sirbrillig** [comment]
I'm going to re-open just to check as @DavidRothstein said he could reproduce this; however, I wonder if maybe he had previously saved contact details? We'll check this during our weekly triage and verify if it's an issue or not.

**Comment by DavidRothstein** [comment]
>In the "phone" section, the OP reports the country flag is not selecting correctly based on where the person is and a number such as +1 234 4556 needs to be entered for the flag to change

No, the bug reported is the opposite of that.  You **can't** type a number such as `+1 234 4556` in order to force the flag to change.

The video above shows the previous behavior, where it worked as designed.  If Brazil was selected and you started typing `+1 234...`, it would automatically switch to the United States.  But now it’s impossible to do that, since if you try to type `+1` while Brazil is selected, it automatically inserts `55` (Brazil's country code) after the `+`, so there's no way to type it, and therefore the country never automatically switches to the United States.

**Comment by jartes** [comment]
I can still replicate this (I re-triaged the issue).
